web/src/main/resources/assets/js/graphhopper/GHRequest.js

```diff
@@ -118,34 +119,60 @@ GHRequest.prototype.getEarliestDepartureTime = function () {
     return undefined;
 };
 
-GHRequest.prototype.initVehicle = function (vehicle) {
-    this.api_params.vehicle = vehicle;
-    if(this.api_params.elevation !== false) {
-        var featureSet = this.features[vehicle];
-        this.api_params.elevation = featureSet && featureSet.elevation;
-    }
-    this.hasTCSupport();
+GHRequest.prototype.setProfile = function (profileName) {
+    this.api_params.profile = profileName;
+};
+
+GHRequest.prototype.getProfile = function () {
+    return this.api_params.profile;
 };
 
-GHRequest.prototype.hasTCSupport = function() {
-   if(this.api_params.turn_costs !== false) {
-      var featureSet = this.features[this.api_params.vehicle];
-      this.api_params.turn_costs = featureSet && featureSet.turn_costs;
-   }
+GHRequest.prototype.setElevation = function (elevation) {
+    this.api_params.elevation = elevation;
```

Comment:
elevation is now treated like a global parameter (instead of being different for every vehicle). for one server this is always the case, but in case there were multiple servers some of which do not support elevation this could be a problem?